### PR TITLE
[AIC-py][editor] test+prod CLI packaging

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[AIC-py][editor] test+prod CLI packaging

This bundles a few simple changes to make CLI installation and usage easier.
The main thing is adding the aiconfig script to the build config, which is tested
using a test pypi package.

- test publish script
- add script to build file that runs cli main
  - had to refactor main slightly to enable this
- open in browser on server start
- only reload server in debug mode
- default prod server mode so you dont have to give it at cli

I built python-aiconfig-test 1.1.22. Install:

`pip3 install --index-url https://test.pypi.org/simple --extra-index-url https://pypi.org/simple python-aiconfig-test==1.1.22 --force`

Run:

`aiconfig edit`

Editor opens in browser

### CODE REVIEW

File: python/src/aiconfig/editor/server/server.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import logging
 from typing import Any, Dict, Type, Union
+import webbrowser
```
Comment (Anpala94) [inline python/src/aiconfig/editor/server/server.py:3]: add to requirements for editor

Comment (Ankush-lastmile) [inline python/src/aiconfig/editor/server/server.py:3]: add to requirements for editor

File: python/src/aiconfig/editor/server/server.py [hunk 2]
```diff
@@ -59,7 +65,8 @@ def run_backend_server(edit_config: EditServerConfig) -> Result[str, str]:
             LOGGER.info("Initialized server state")
             debug = edit_config.server_mode in [ServerMode.DEBUG_BACKEND, ServerMode.DEBUG_SERVERS]
             LOGGER.info(f"Running in {edit_config.server_mode} mode")
-            app.run(port=edit_config.server_port, debug=debug, use_reloader=True)
+            use_reloader = edit_config.server_mode in [ServerMode.DEBUG_BACKEND, ServerMode.DEBUG_SERVERS]
```
Comment (Ankush-lastmile) [inline python/src/aiconfig/editor/server/server.py:68]: what does use_reloader mean in this context?

**Comment by Ankush-lastmile** [comment]
> The main thing is adding the aiconfig script to the build config, which is tested
using a test pypi package.

Missing a testplan

**Comment by Ankush-lastmile** [comment]
> This bundles a few simple changes to make CLI installation and usage easier.

Can you elaborate on what this means or what this change does to enable that? this description is unclear what the changes are

File: python/pyproject.toml [hunk 3]
```diff
@@ -23,6 +23,9 @@ dependencies = {file = ["requirements.txt"]}
 "Homepage" = "https://github.com/lastmile-ai/aiconfig"
 "Bug Tracker" = "https://github.com/lastmile-ai/aiconfig/issues"
 
+[project.scripts]
+aiconfig = "aiconfig.scripts.aiconfig_cli:main"
```
Comment (Ankush-lastmile) [inline python/pyproject.toml:27]: is this tested?

**Review by Ankush-lastmile — CHANGES_REQUESTED** [review]
requesting changes so that comments get addressed. Code lgtm besides those comments

File: python/src/aiconfig/editor/server/server.py (hunk 1 above)
Comment (jonathanlastmileai) [reply python/src/aiconfig/editor/server/server.py:3]: standard lib

https://docs.python.org/3/library/webbrowser.html

File: python/src/aiconfig/editor/server/server.py (hunk 2 above)
Comment (jonathanlastmileai) [reply python/src/aiconfig/editor/server/server.py:68]: > Enable automatic reloads on code changes.
--

https://pgjones.gitlab.io/hypercorn/how_to_guides/configuring.html


Unclear if it actually works tough

File: python/pyproject.toml (hunk 3 above)
Comment (jonathanlastmileai) [reply python/pyproject.toml:27]: yes, added details in test plan

**Comment by jonathanlastmileai** [comment]
> > The main thing is adding the aiconfig script to the build config, which is tested
> > using a test pypi package.
> 
> Missing a full testplan, just describing it is not enough

added commands for testing

**Comment by jonathanlastmileai** [comment]
> > This bundles a few simple changes to make CLI installation and usage easier.
> 
> Can you elaborate on what this means or what this change does to enable that? this description is unclear what the changes are

updated description

File: python/pypipublish-test.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#! /bin/zsh
```
Comment (Ankush-lastmile) [inline python/pypipublish-test.sh:1]: should this be inside the src dir?  Wouldn't it make more sense for this script to be either in the root of aiconfig or aiconfig/scripts

File: python/src/aiconfig/editor/server/server.py (hunk 2 above)
Comment (Ankush-lastmile) [reply python/src/aiconfig/editor/server/server.py:68]: >Unclear if it actually works tough

If its unclear if it works then either that should be tested or removed from the pr

Comment (Ankush-lastmile) [reply python/src/aiconfig/editor/server/server.py:68]: > Enable automatic reloads on code changes.

I'm missing the context on why server_mode is the option you pass into use_reloader. Could you explain what is going on here?

Comment (jonathanlastmileai) [reply python/src/aiconfig/editor/server/server.py:68]: 3 things:

1. I mixed up this and V2 server which uses hypercorn. Reloader does work with flask (it doesn't seem to work with either uvicorn or hypercorn).

2. This change doesn't add the use of reloader, it just turns _off_ reloader if we're not in a debug mode. This change is clearly correct: it just changes the use_reloader condition based on server mode. 
  b) However, I just noticed that i already gave this condition a name, `debug` and should reuse that. 
3. The motivation for this logic is so you can reload the server on change if and only if you're debugging.

File: python/pypipublish-test.sh (hunk 4 above)
Comment (jonathanlastmileai) [reply python/pypipublish-test.sh:1]: I'll just remove this for now

**Review by Ankush-lastmile — APPROVED** [review]
discussed offline. @jonathanlastmileai tested these changes successfully locally

**Review by saqadri — APPROVED** [review]

**Pull request merged**